src/main/java/com/textmessenger/service/MessageServiceJpa.java

```diff
@@ -0,0 +1,46 @@
+package com.textmessenger.service;
+
+import com.textmessenger.model.Message;
+import com.textmessenger.repository.MessageRepository;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class MessageServiceJpa implements MessageService {
+
+  private final MessageRepository messageRepository;
+
+  public MessageServiceJpa(MessageRepository messageRepository) {
+    this.messageRepository = messageRepository;
+  }
+
+  @Override
+  @Transactional
+  public long createMessage(Message message) {
+    return messageRepository.save(message).getId();
+  }
+
+  @Override
+  public Message readMessage(long id) {
+    return messageRepository.getOne(id);
+  }
+
+  @Override
+  @Transactional
+  public void updateMessage(long id, Message message) {
```

Comment:
You can change the interface. updateMessage doesn't need an ID, it can be taken from incoming Message